front/src/common/ChangeLanguageModal.tsx

```diff
@@ -10,61 +10,71 @@ import { supportedLngs } from 'i18n';
 
 import { useModal } from './BootstrapSNCF/ModalSNCF';
 
-const SortedLanguages = () => {
-  const { t, i18n } = useTranslation();
+// We don't use `t` cause language names are fixed
+export const languageName = (lng: string) => {
+  switch (lng) {
+    case 'de':
+      return 'Deutsch';
+    case 'en':
+      return 'English';
+    case 'fr':
+      return 'Français';
+    default:
+      return null;
+  }
+};
+
+const languageSvgFlag = (lng: string) => {
+  switch (lng) {
+    case 'de':
+      return <DE title="Deutsch" />;
+    case 'en':
+      return <GB title="English" />;
+    case 'fr':
+      return <FR title="Français" />;
+    default:
+      return null;
+  }
+};
+
+const Languages = () => {
+  const { i18n } = useTranslation();
   const { closeModal } = useModal();
 
   const changeLanguage = (lng: string) => {
     i18n.changeLanguage(lng);
     closeModal();
   };
 
-  const formatLanguageWithFlag = (lng: string) => {
-    const title = t(`nav-bar.language.${lng}`);
-    switch (lng) {
-      case 'de':
-        return <DE title={title} />;
-      case 'en':
-        return <GB title={title} />;
-      case 'fr':
-        return <FR title={title} />;
-      default:
-        return null;
-    }
-  };
-
-  const availablesLanguages = supportedLngs.map((lng) => ({
-    key: lng,
-    value: t(`nav-bar.language.${lng}`),
-  }));
-  availablesLanguages.sort((a, b) => a.value.localeCompare(b.value));
   return (
     <>
-      {availablesLanguages.map((lng) => (
+      {supportedLngs.map((language_code) => (
```

Comment:
Ah, style nit: we use camelCase instead of snake_case

Reply:
Why is this not checked by prettier or eslint?

Reply:
Probably because no-one bothered… I can see one other reason though: we manipulate types coming from the API in a lot of places and these all use snake_case. Maybe we can still try to enable it for variable names only.